test/hdimage-fail9.config

```diff
@@ -0,0 +1,27 @@
+image test.hdimage {
+	hdimage {
+		align = 1M
+	}
+	partition primary1 {
+		image = "part1.img"
+		partition-type = 0x83
+	}
+	partition primary2 {
+		image = "part1.img"
+		partition-type = 0x83
+	}
+	partition primary3 {
+		image = "part1.img"
+		partition-type = 0x83
+	}
+	partition primary4 {
+		image = "part1.img"
+		partition-type = 0x83
+	}
+	partition primary5 {
+		image = "part1.img"
+		partition-type = 0x83
+		/* part4 is implicitly extended -> too many primary entries */
```

Comment:
Why does this test fail? Shouldn't primary partition 3 be the extended partition in this case, containing partition 3 and 4? :thinking:

Reply:
No, if `extended-partition` is not specified in the `hdimage` section, then it defaults to 4. Which means `primary4` would become the first extended (or, to be precise, logical within extended) partition, and the MBR will be fully occupied by three primary entries `primary1-3` and one extended entry.